src/libudev/libudev-monitor.c

```diff
@@ -206,12 +206,10 @@ static int udev_monitor_receive_sd_device(struct udev_monitor *udev_monitor, sd_
                 for (;;) {
                         /* Wait for next message */
                         r = fd_wait_for_event(device_monitor_get_fd(udev_monitor->monitor), POLLIN, 0);
-                        if (r < 0) {
-                                if (IN_SET(r, -EINTR, -EAGAIN))
-                                        continue;
-
+                        if (r == -EINTR)
+                                continue;
```

Comment:
actually, checking for EAGAIN might not be the worst thing, given the comment at the end of the poll() man page:

>  On  some other UNIX systems, poll() can fail with the error EAGAIN if the system fails to allocate kernel-internal resources, rather than ENOMEM as Linux does.  POSIX permits this behavior.  Portable programs may wish to check for EAGAIN and loop, just as with EINTR.

Reply:
This is for `poll()`, not for `ppoll()`. And the man page said that `ppoll()` is linux specific.